statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,612 @@
+            # accumulate number of times cluster i was resampled
+            clust_count += np.bincount(ii_resample,
+                                       minlength=max(self.sclust)+1)
+
+        w *= clust_count[self.sclust]
+        return w
+
+    def _mean_bootstrap_weight(self, bsn):
+        """
+        Computes mean bootstrap replicate weight
+
+        Parameters
+        ----------
+        bsn : integer
+            Mean bootstrap averages the number of resampled clusters over bsn
+        Returns
+        -------
+        w : ndarray
+            Augmented weight
+        """
+        clust_count = np.zeros(self.design.nclust)
+        w = self.weights.copy()
+        # for each replicate, I accumulate bsn number of times?
+        for b in range(bsn):
+            for s in range(self.nstrat):
+                w[self.strat == s] *= ((float(self.ncs[s] - 1) /
+                                        self.ncs[s])**(1/bsn))
+                # If there is only one or two clusters then weights wont change
+                if (self.ncs[s] == 1 or self.ncs[s] == 2):
+                    continue
+                # resample array of clusters in strata s
+                ii_resample = np.random.choice(self.ii[s], size=(self.ncs[s]-1))
+                # accumulate number of times cluster i was resampled
+                clust_count += np.bincount(ii_resample,
+                                           minlength=max(self.sclust)+1)
+        # avg number of times cluster i was resampled
+        clust_count /= bsn
+        # augment weights
+        w *= clust_count[self.sclust]
+        return w
+
+
+class SurveyStat(SurveyDesign):
+    """
+    Estimation and inference for summary statistics in complex surveys.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+
+    Attributes
+    ----------
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    vcov : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+
+    def __init__(self, design, mse=False):
+        self.design = design
+        self.mse = mse
+
+    def _bootstrap(self, replicates=None, bsn=None):
+        """
+        Calculates bootstrap standard errors
+
+        Parameters
+        ----------
+        stat : object
+            Object of class SurveyMean, SurveyTotal, SurveyPercentile, etc
+        replicates : integer
+            The number of replicates that the user wishes to specify
+
+        Returns
+        -------
+        est : ndarray
+            The point estimates of the statistic, calculated on the columns
+            of data.
+        vcov : ndarray
+            The variance-covariance of the estimates.
+        """
+        est = self._stat(self.design.weights)
+
+        jdata = []
+        self.w_li = []
+        for i in range(replicates):
+            w = self.design.get_rep_weights(i, bsn=bsn)
```

Comment:
related to my comments on where `se_method` is defined as argument by user and attached as attribute
calling `get_rep_weights` here assumes that `design.se_method == 'boot'`, AFAICS.
This is more difficult to debug, because the behavior depends on the calling sequence (how we got here) and that the "hidden" attribute in design is consistent with this.

Instead, I think it would be cleaner that `design.get_rep_weights` has a se_method (or cov_method) argument.
AFAICS, nothing in the design class assumes that the se_method is fixed.